executor/builder.go

```diff
@@ -4416,9 +4416,7 @@ func (b *executorBuilder) validCanReadTemporaryTable(tbl *model.TableInfo) error
 	// Some tools like dumpling use history read to dump all table's records and will be fail if we return an error.
 	// So we do not check SnapshotTS here
 
-	sessionVars := b.ctx.GetSessionVars()
-
-	if sessionVars.TxnCtx.IsStaleness || b.explicitStaleness {
+	if b.isStaleness {
```

Comment:
`TxnCtx.IsStaleness` is not deprecated completely, I think. Maybe remove it in other PRs.